tests/base_scraper/test_base_scraper.py

```diff
@@ -74,5 +74,39 @@ def test_notimplementedexceptions(self):
         self.assertRaises(mozdownload.NotImplementedError,
                           scraper.build_filename, 'invalid binary')
 
+    def test_authentication(self):
+        """testing authentication"""
+        username = 'mozilla'
+        password = 'mozilla'
+        basic_auth_url = 'http://mozqa.com/data/mozqa.com/http_auth/basic/'
+        no_auth_url = 'https://ci.mozilla.org/' #does not need auth
+
+        # test with invalid authentication
+        # check if exception is thrown
```

Comment:
The second comment line you can remove. The call to `assertRaises` makes that clear.